controllers/controllers/workloads/cfapp_controller.go

```diff
@@ -181,6 +191,82 @@ func getDesiredInstanceCount(processType string) int {
 	return 0
 }
 
+func (r *CFAppReconciler) addFinalizer(ctx context.Context, cfApp *workloadsv1alpha1.CFApp) error {
+	if controllerutil.ContainsFinalizer(cfApp, finalizerName) {
+		return nil
+	}
+
+	originalCFApp := cfApp.DeepCopy()
+	controllerutil.AddFinalizer(cfApp, finalizerName)
+
+	err := r.Client.Patch(ctx, cfApp, client.MergeFrom(originalCFApp))
+	if err != nil {
+		r.Log.Error(err, fmt.Sprintf("Error adding finalizer to CFApp/%s", cfApp.Name))
+		return err
+	}
+
+	r.Log.Info(fmt.Sprintf("Finalizer added to CFApp/%s", cfApp.Name))
+	return nil
+}
+
+func isFinalizing(cfApp *workloadsv1alpha1.CFApp) bool {
+	return cfApp.ObjectMeta.DeletionTimestamp != nil && !cfApp.ObjectMeta.DeletionTimestamp.IsZero()
+}
+
+func (r *CFAppReconciler) finalizeCFApp(ctx context.Context, cfApp *workloadsv1alpha1.CFApp) (ctrl.Result, error) {
+	r.Log.Info(fmt.Sprintf("Reconciling deletion of CFApp/%s", cfApp.Name))
+
+	if !controllerutil.ContainsFinalizer(cfApp, finalizerName) {
+		return ctrl.Result{}, nil
+	}
+
+	cfRoutes, err := r.getCFRoutes(ctx, cfApp.Name, cfApp.Namespace)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	for _, cfRoute := range cfRoutes {
+		originalCFRoute := cfRoute.DeepCopy()
+		if cfRoute.Spec.Destinations != nil {
+			for idx, destination := range cfRoute.Spec.Destinations {
+				if destination.AppRef.Name == cfApp.Name {
+					r.Log.Info(fmt.Sprintf("Removing destination for cfapp %s from cfroute %s", cfApp.Name, cfRoute.Name))
+					cfRoute.Spec.Destinations[idx] = cfRoute.Spec.Destinations[len(cfRoute.Spec.Destinations)-1]
+					cfRoute.Spec.Destinations = cfRoute.Spec.Destinations[:len(cfRoute.Spec.Destinations)-1]
+				}
+			}
+		}
+		err = r.Client.Patch(ctx, &cfRoute, client.MergeFrom(originalCFRoute))
+		if err != nil {
+			r.Log.Error(err, "failed to patch cfRoute to remove a destination")
+			return ctrl.Result{}, err
+		}
+	}
```

Comment:
I think it would improve readability and make it more clear what's going on here to move this into another function called like `removeRouteDestinations` or something

Reply:
Thanks for the review. I have addressed this